main.tf

```diff
@@ -185,10 +185,24 @@ resource "azurerm_kubernetes_cluster" "main" {
     network_plugin     = var.network_plugin
     dns_service_ip     = var.net_profile_dns_service_ip
     docker_bridge_cidr = var.net_profile_docker_bridge_cidr
+    load_balancer_sku  = var.load_balancer_sku
     network_policy     = var.network_policy
     outbound_type      = var.net_profile_outbound_type
     pod_cidr           = var.net_profile_pod_cidr
     service_cidr       = var.net_profile_service_cidr
+
+    dynamic "load_balancer_profile" {
+      for_each = var.enable_load_balancer_profile == true && var.load_balancer_sku == "standard" ? ["load_balancer_profile"] : []
```

Comment:
Since the `var.enable_load_balancer_profile` should not be `null`, could we simplify the expression to the following code?:

```hcl
for_each = var.enable_load_balancer_profile && var.load_balancer_sku == "standard" ? ["load_balancer_profile"] : []
```